Update torch.autograd.graph logging to not print out grad_output

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116710
* __->__ #116523

Instead of printing the tensor's data print the dtype and shape metadata of the tensor.
```
Executing: <VarMeanBackward0 object at 0x1352d0e20> with grad_outputs: [None,f32[]]
```
This is important in order to avoid doing a cuda sync and also useful to reduce verbosity.

soulitzer committed:
Update torch.autograd.graph logging to not print out grad_output

[ghstack-poisoned]

albanD commented on torch/autograd/graph.py:
```diff
@@ -659,7 +659,7 @@ def iter_graph(roots):
     def prehook(grad_output):
         node = torch._C._current_autograd_node()
-        log_str = f"Executing: {node} with grad_output: {grad_output}"
+        log_str = f"Executing: {node}"
```
If this is too verbose, feel free to only print size to share some information.

albanD reviewed (approved):
SGTM

soulitzer committed:
Update on "Update torch.autograd.graph logging to not print out grad_output"




[ghstack-poisoned]

albanD commented on torch/autograd/graph.py:
```diff
@@ -657,9 +657,18 @@ def iter_graph(roots):
+
+        if t is None:
+            return "None"
+        return f"{dtype_abbrs[t.dtype]}[{', '.join(map(str, t.shape))}]"
```
Is that a usual format used in other places?

soulitzer replied:
Yeah, fx and logging tensor also use this format

albanD reviewed (commented):
Change sounds good but needs rebase!

soulitzer committed:
Update on "Update torch.autograd.graph logging to not print out grad_output"


Instead of printing the tensor's data print the dtype and shape metadata of the tensor.
```
Executing: <VarMeanBackward0 object at 0x1352d0e20> with grad_outputs: [None,f32[]]
```
This is important in order to avoid doing a cuda sync and also useful to reduce verbosity.


[ghstack-poisoned]

soulitzer committed:
Update on "Update torch.autograd.graph logging to not print out grad_output"


Instead of printing the tensor's data print the dtype and shape metadata of the tensor.
```
Executing: <VarMeanBackward0 object at 0x1352d0e20> with grad_outputs: [None,f32[]]
```
This is important in order to avoid doing a cuda sync and also useful to reduce verbosity.


[ghstack-poisoned]

soulitzer committed:
Update on "Update torch.autograd.graph logging to not print out grad_output"


Instead of printing the tensor's data print the dtype and shape metadata of the tensor.
```
Executing: <VarMeanBackward0 object at 0x1352d0e20> with grad_outputs: [None,f32[]]
```
This is important in order to avoid doing a cuda sync and also useful to reduce verbosity.


[ghstack-poisoned]

soulitzer commented:
@pytorchbot merge
